### DISCUSSION TITLE
Integration tests failing

### DISCUSSION THREAD

**Discussion by tkennes** [cloud-carbon-footprint/cloud-carbon-footprint#1172] [Q&A] [answered]
HI,

I'm running consistenly into issues with the front-end integration tests, even though I only make changes in the backend. I'm not sure why these occur though, can you help me out here? Do I need to add some settings or provide some more manual data?

Screenshot:
![1](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/assets/21101093/41021bea-80f4-4c81-8b06-5934a8cfc21a)

The error:
 1) AssertionError: expected false to be truthy
@cloud-carbon-footprint/integration-tests:       Browser: Chrome 114.0.5735.133 / macOS 10.15.7
@cloud-carbon-footprint/integration-tests:       Screenshot:
@cloud-carbon-footprint/integration-tests:    /Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/screenshots/2023-06-23_10-58-40/test-10/Chrome_114.0.5735.133_macOS_10.15.7/errors/1.png
@cloud-carbon-footprint/integration-tests:          32 |})
@cloud-carbon-footprint/integration-tests:          33 |
@cloud-carbon-footprint/integration-tests:          34 |test('card components render with correct data when app
@cloud-carbon-footprint/integration-tests:       loads', async (t) => {
@cloud-carbon-footprint/integration-tests:          35 |  // commented out because the error message is not always
@cloud-carbon-footprint/integration-tests:       present
@cloud-carbon-footprint/integration-tests:          36 |  // await t.expect(page.errorMessage.exists).ok()
@cloud-carbon-footprint/integration-tests:        > 37 |  await t.expect(page.lastThirtyDayTotal.exists).ok()
@cloud-carbon-footprint/integration-tests:          38 |  await t.expect(page.projectedThirtyDayTotal.exists).ok()
@cloud-carbon-footprint/integration-tests:          39 |  await t.expect(page.forecastEquivalencyCard.exists).ok()
@cloud-carbon-footprint/integration-tests:          40 |  await t.expect(page.treeSeedlingsGrown.exists).ok()
@cloud-carbon-footprint/integration-tests:          41 |  await t.expect(page.costSavingsPerMonth.exists).ok()
@cloud-carbon-footprint/integration-tests:          42 |})
@cloud-carbon-footprint/integration-tests:          at <anonymous>
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:37:50)
@cloud-carbon-footprint/integration-tests:          at asyncGeneratorStep
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:6:259)
@cloud-carbon-footprint/integration-tests:          at _next
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:6:597)
@cloud-carbon-footprint/integration-tests:          at <anonymous>
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:6:762)
@cloud-carbon-footprint/integration-tests:          at <anonymous>
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:6:503)
@cloud-carbon-footprint/integration-tests:          at <anonymous>
@cloud-carbon-footprint/integration-tests:       (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:34:5)
@cloud-carbon-footprint/integration-tests:  ✓ table components render with correct data when app loads
@cloud-carbon-footprint/integration-tests:  ✖ toggle changes unit of measure (screenshots:
@cloud-carbon-footprint/integration-tests:  /Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/screenshots/2023-06-23_10-58-40/test-12/Chrome_114.0.5735.133_macOS_10.15.7/errors/1.png)
@cloud-carbon-footprint/integration-tests:    1) TypeError: Cannot read properties of undefined (reading
@cloud-carbon-footprint/integration-tests:       'textContent')
@cloud-carbon-footprint/integration-tests:       Browser: Chrome 114.0.5735.133 / macOS 10.15.7
@cloud-carbon-footprint/integration-tests:       Screenshot:
@cloud-carbon-footprint/integration-tests:    /Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/screenshots/2023-06-23_10-58-40/test-12/Chrome_114.0.5735.133_macOS_10.15.7/errors/1.png
@cloud-carbon-footprint/integration-tests:          50 |  //check projected totals
@cloud-carbon-footprint/integration-tests:          51 |  // commented out because the error message is not always
@cloud-carbon-footprint/integration-tests:       present
@cloud-carbon-footprint/integration-tests:          52 |  // await t.expect(page.errorMessage.exists).ok()
@cloud-carbon-footprint/integration-tests:          53 |
@cloud-carbon-footprint/integration-tests:          54 |  await t
@cloud-carbon-footprint/integration-tests:        > 55 |    .expect(page.unitOfMeasureLastThirtyDayTotal.textContent)
@cloud-carbon-footprint/integration-tests:          56 |    .eql('Metric Tons CO2e')
@cloud-carbon-footprint/integration-tests:          57 |  await t
@cloud-carbon-footprint/integration-tests:          58 |
@cloud-carbon-footprint/integration-tests:       .expect(page.unitOfMeasureProjectedThirtyDayTotal.textContent)
@cloud-carbon-footprint/integration-tests:          59 |    .eql('Metric Tons CO2e')
@cloud-carbon-footprint/integration-tests:          60 |  await
@cloud-carbon-footprint/integration-tests:       t.expect(page.co2eSavingsLastThirtyDayTotal.exists).ok()
@cloud-carbon-footprint/integration-tests:          at <anonymous>
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:55:50)
@cloud-carbon-footprint/integration-tests:          at asyncGeneratorStep
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:6:259)
@cloud-carbon-footprint/integration-tests:          at _next
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:6:597)
@cloud-carbon-footprint/integration-tests:          at <anonymous>
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:6:762)
@cloud-carbon-footprint/integration-tests:          at <anonymous>
@cloud-carbon-footprint/integration-tests:    (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:6:503)
@cloud-carbon-footprint/integration-tests:          at <anonymous>
@cloud-carbon-footprint/integration-tests:       (/Users/tomkennes/Documents/cloud-carbon-footprint/packages/integration-tests/tests/recommendations.test.js:49:5)
@cloud-carbon-footprint/integration-tests:  2/12 failed (1m 45s)
@cloud-carbon-footprint/integration-tests:  Warnings (1):

**Comment by 4upz** [comment, marked as answer]
@tkennes Unfortunately, we've been having problems with the integration tests recently where they can be a bit inconsistent. We're currently in the process of updating them so this no longer happens. 

As a first precaution, make sure you have no other instance of the app running locally while the integration tests are running. If you have already ensure this and they are still failing, then you can do the following when committing: 

- Use the pre-commit hook to check if all other test suites besides the integrations tests are passing, then use the `--no-verify` flag to bypass the hook and commit you changes.
- Remove `yarn headless` from the precommit command in the integration test's package.json file (`packages/integration-tests/package.json` Line 12)

**Reply by tkennes** [reply]
Thanks for picking it up!
